### ISSUE TITLE
Crafting Clean and Maintainable APIs with Django REST Framework

### ISSUE THREAD

**Issue by mathematical-guy** [djangoindia/talks#3]
### Title

Crafting Clean and Maintainable APIs with Django REST Framework

### Describe your Talk

Hello Django peeps 😎, this talk will cover about

 - What is DRF and it's basic overview ? 💥
 - CRUD Operations made easy with APIView, GenericView and ViewSets 💥
 - Validation, Searching, filtering and pagination 💥
 - How to craft industry standard URL ? 💥
 - Authentication / Authorization handling in DRF 💥
 - Rendering standard response. 💥

### Pre-requisites & reading material

 - Basic familiarity with Django framework ✅
 - Having knowledge of HTTP methods ✅
 - Understanding of what API is (will try to cover it in talk) ✅

### Time required for the talk

30 Minutes in total

### Link to slides/demos

**PPT Link:** https://docs.google.com/presentation/d/1f3DjzXdlM_fIEvImzC4wziUwXsjDUSI2nGpiA0TWP6k/edit#slide=id.g31a8993d843_0_1974

### About you

3+ Years of Experience in developing dynamic web applications. 💪
Expertise in **Python/Django**, with strong skills in Flask, FastAPI, and REST API development. 😎
Proficient in Database Management: PostgreSQL, MongoDB, and SQLite. ⭐
Frontend Development: HTML, CSS, JavaScript, ReactJS. 💙
Cloud Experience: AWS (EC2, RDS) with a focus on scalable deployments. ⛅
Passionate Problem-Solver dedicated to leveraging technology to drive innovation. ❓
Nerd, Physics & Mathematics lover. 🤓

### Availability

07/12/2024

**Comment by DevilsAutumn** [comment]
Hi @mathematical-guy ! Thanks for submitting this talk, looks like a good topic for beginners.
Do you have any ppt to present along with the talk?

**Comment by mathematical-guy** [comment]
Hey @DevilsAutumn 
Yes, I was about to add PPT link, but it's under progress now.

**Comment by DevilsAutumn** [comment]
Sounds good ! @mathematical-guy 👍

**Comment by DevilsAutumn** [comment]
@mathematical-guy Do you have any ppt ready now? We're very close to the meetup and need to finalize the speakers.

**Comment by mathematical-guy** [comment]
Yes @DevilsAutumn , making few minor changes will get updated soon in the above link

**Comment by mathematical-guy** [comment]
Yes @DevilsAutumn , making few minor changes will get updated soon in the above link
